### PR TITLE
Add fastseg-small, fastseg-large models

### PR DESCRIPTION
**Pull request by jkamelin**

### CODE REVIEW

File: models/public/fastseg-large/fastseg-large.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+Float values, which represent scores of a predicted class for each image pixel. Name: `output0`, shape: [1x19x1024x2048] in [BxNxHxW] format, where
+
+    - B - batch size
+    - N - number of classes
+    - H - image height
+    - W - image width
+
+### Converted Model
+
+Float values, which represent scores of a predicted class for each image pixel. Name: `output0`, shape: [1x19x1024x2048] in [BxNxHxW] format, where
```
Comment (eaidova) [inline models/public/fastseg-large/fastseg-large.md:62]: please provide info about dataset across which model predict labels

File: models/public/fastseg-small/pre-convert.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+BACKBONE = 'MobileV3Small'
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('input_dir', type=Path)
+    parser.add_argument('output_dir', type=Path)
+    args = parser.parse_args()
+
+    subprocess.run([sys.executable, '--',
```
Comment (eaidova) [inline models/public/fastseg-small/pre-convert.py:31]: do you tested this changes on windows? as I know there can be some issues with subprocess running?

Comment (jkamelin) [reply models/public/fastseg-small/pre-convert.py:31]: yes, this script works on windows

File: models/public/fastseg-large/fastseg-large.md (hunk 1 above)
Comment (jkamelin) [reply models/public/fastseg-large/fastseg-large.md]: done

File: models/public/fastseg-large/fastseg-large.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    - H - image height
+    - W - image width
+
+   Expected color order: BGR.
+
+## Output
+
+### Original Model
+
+Float values, which represent scores of a predicted class for each image pixel. The model was trained on Cityscapes dataset with 19 categories of objects. Name: `output0`, shape: [1, 19, 1024, 2048] in [B, N, H, W] format, where
```
Comment (eaidova) [inline models/public/fastseg-large/fastseg-large.md:53]:
```suggestion
Float values, which represent scores of a predicted class for each image pixel. The model was trained on Cityscapes dataset with 19 categories of objects. Name: `output0`, shape: `1, 19, 1024, 2048` in `B, N, H, W` format, where
```

File: models/public/fastseg-large/fastseg-large.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+    - B - batch size
+    - N - number of classes
+    - H - image height
+    - W - image width
+
+### Converted Model
+
+Float values, which represent scores of a predicted class for each image pixel. The model was trained on Cityscapes dataset with 19 categories of objects. Name: `output0`, shape: [1, 19, 1024, 2048] in [B, N, H, W] format, where
```
Comment (eaidova) [inline models/public/fastseg-large/fastseg-large.md:62]:
```suggestion
Float values, which represent scores of a predicted class for each image pixel. The model was trained on Cityscapes dataset with 19 categories of objects. Name: `output0`, shape: `1, 19, 1024, 2048` in `B, N, H, W` format, where
```

Comment (eaidova) [reply models/public/fastseg-large/fastseg-large.md:62]: and the same for inputs, according to internal discussion

File: models/public/fastseg-large/fastseg-large.md (hunk 3 above)
Comment (jkamelin) [reply models/public/fastseg-large/fastseg-large.md]: done

File: models/public/fastseg-large/fastseg-large.md (hunk 4 above)
Comment (jkamelin) [reply models/public/fastseg-large/fastseg-large.md]: done

**Review by eaidova — APPROVED** [review]

File: models/public/fastseg-large/fastseg-large.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
+# fastseg-large
+
+## Use Case and High-Level Description
+
+fastseg-large is an accurate real-time semantic segmentation model. For details see [repository](https://github.com/ekzhang/fastseg).
```
Comment (vladimir-dudnik) [inline models/public/fastseg-large/fastseg-large.md:5]:
```suggestion
fastseg-large is an accurate real-time semantic segmentation model, pretrained on [Cityscapes](https://www.cityscapes-dataset.com) dataset for 19 object classes, see Cityscapes classes [defintion](https://www.cityscapes-dataset.com/dataset-overview). The model built on MobileNetV3 large backbone and modified segmentation head based on LR-ASPP. This model can be used for efficient segmentation on a variety of real-world street images. For details see [repository](https://github.com/ekzhang/fastseg).
```

File: models/public/fastseg-small/fastseg-small.md [hunk 6]
```diff
@@ -0,0 +1,96 @@
+# fastseg-small
+
+## Use Case and High-Level Description
+
+fastseg-small is an accurate real-time semantic segmentation model. For details see [repository](https://github.com/ekzhang/fastseg).
```
Comment (vladimir-dudnik) [inline models/public/fastseg-small/fastseg-small.md:5]:
```suggestion
fastseg-small is an accurate real-time semantic segmentation model, pretrained on [Cityscapes](https://www.cityscapes-dataset.com) dataset for 19 object classes, see Cityscapes classes [defintion](https://www.cityscapes-dataset.com/dataset-overview). The model built on MobileNetV3 small backbone and modified segmentation head based on LR-ASPP. This model can be used for efficient segmentation on a variety of real-world street images. For details see [repository](https://github.com/ekzhang/fastseg).
```

**Review by vladimir-dudnik — COMMENTED** [review]
are these models work in our existing segmentation demos? we need this and also need to add models to tests.

**Review by vladimir-dudnik — COMMENTED** [review]
Is it possible to provide Citiscapes labels for 19 classes, models were trained for?

File: models/public/fastseg-large/model.yml [hunk 7]
```diff
@@ -0,0 +1,74 @@
+# Copyright (c) 2020 Intel Corporation
```
Comment (eizamaliev) [inline models/public/fastseg-large/model.yml:1]:
```suggestion
# Copyright (c) 2021 Intel Corporation
```

File: models/public/fastseg-small/model.yml [hunk 8]
```diff
@@ -0,0 +1,74 @@
+# Copyright (c) 2020 Intel Corporation
```
Comment (eizamaliev) [inline models/public/fastseg-small/model.yml:1]:
```suggestion
# Copyright (c) 2021 Intel Corporation
```

File: models/public/fastseg-small/pre-convert.py [hunk 9]
```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2020 Intel Corporation
```
Comment (eizamaliev) [inline models/public/fastseg-small/pre-convert.py:3]:
```suggestion
# Copyright (c) 2021 Intel Corporation
```

**Review by eizamaliev — CHANGES_REQUESTED** [review]

**Review by vladimir-dudnik — APPROVED** [review]
except @eizamaliev notice regarding years in copyrights, other seems ok for me

File: models/public/fastseg-small/pre-convert.py (hunk 9 above)
Comment (jkamelin) [reply models/public/fastseg-small/pre-convert.py]: done

**Comment by eaidova** [comment]
@jkamelin please rebase your PR

**Review by eizamaliev — APPROVED** [review]

File: models/public/fastseg-small/pre-convert.py [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('input_dir', type=Path)
+    parser.add_argument('output_dir', type=Path)
+    args = parser.parse_args()
+
+    subprocess.run([sys.executable, '--',
+        str(args.input_dir / 'model/onnx_export.py'),
+        '{}'.format(args.output_dir / 'fastseg-small.onnx'),
+        '--checkpoint={}'.format(args.output_dir / 'mobilev3small-lraspp-f128-a39a1e4b.pt'),
```
Comment (IRDonch) [inline models/public/fastseg-small/pre-convert.py:34]: I think this was supposed to be `args.input_dir`. Please submit a fix.

**Pull request merged**